app/controllers/users_controller.rb

```diff
@@ -2,7 +2,11 @@ class UsersController < ApplicationController
   before_action :check_account_and_redirect, only: :new
   before_action :is_admin_role?, only: %i(index)
   def index
-    @pagy, @users = pagy User.order_by_name, items: Settings.users_per_page
+    if params[:status] == "banned"
+      @pagy, @users = pagy User.banned, items: Settings.users_per_page
+    else
+      @pagy, @users = pagy User.order_by_name, items: Settings.users_per_page
+    end
```

Comment:
```suggestion
      @pagy, @users = pagy User.order_by_name.where(status: params[:status]), items: Settings.users_per_page
```
Nếu là anh thì anh sẽ viết như này
Chỗ `where(status: params[:status])` em viết scope thêm đk `if params[:status].present?` nhé